tests/publish-dsl.nf

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env nextflow
+/*
+ * Copyright 2013-2024, Seqera Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+process align {
+  input:
+  val(x)
+
+  output:
+  path("*.bam")
+  path("${x}.bai")
+
+  """
+  echo ${x} > ${x}.bam
+  echo ${x} | rev > ${x}.bai
+  """
+}
+
+process my_combine {
+  input:
+  path(bamfile)
+  path(baifile)
+
+  output:
+  path 'result.txt'
+
+  """
+  cat $bamfile > result.txt
+  cat $baifile >> result.txt
+  """
+}
+
+process foo {
+  output:
+  path 'xxx'
+
+  '''
+  mkdir xxx
+  touch xxx/A
+  touch xxx/B
+  touch xxx/C
+  '''
+}
+
+workflow {
+  def input = Channel.of('alpha','beta','delta')
+  align(input)
+
+  def bam = align.out[0].toSortedList { it.name }
+  def bai = align.out[1].toSortedList { it.name }
+  my_combine( bam, bai )
+  my_combine.out.view{ it.text }
+
+  foo()
+
+  publish:
+  align.out[0]    >> 'data'
+  align.out[1]    >> 'data'
+  my_combine.out  >> 'more/data'
+  foo.out         >> 'data'
+}
+
+output {
+  directory 'results'
```

Comment:
I think `path` can be used instead of `directory` both for internal and `process output` consistency

Reply:
I considered the same, but there is an important difference with `directory` -- whereas with other publish options the rule overrides the generic option, the `path` in a rule is appended to the top-level `path` i.e. `directory`. So I think the different name is actually helpful to denote the different semantics